src/proto/h1/io.rs

```diff
@@ -419,6 +424,24 @@ impl<T: AsRef<[u8]>> Cursor<T> {
 }
 
 impl Cursor<Vec<u8>> {
+    /// If we've advanced the position a bit in this cursor, and wish to
+    /// extend the underlying vector, we may wish to unshift the "read" bytes
+    /// off, and move everything else over.
+    fn maybe_unshift(&mut self, additional: usize) {
```

Comment:
I'm not sure this method name really fits the method, but I have no alternative to offer.

It took me a hot second to understand that `additional` is the threshold we check to decide whether or not to drain the leading bytes of the buffer.

Reply:
I agree, coming up with a name was hard.